lib/danger/danger_core/plugins/dangerfile_bitbucket_server_plugin.rb

```diff
@@ -0,0 +1,179 @@
+  # @tags core, bitbucket_server
+  #
+  class DangerfileBitbucketServerPlugin < Plugin
+    # So that this init can fail.
+    def self.new(dangerfile)
+      return nil if dangerfile.env.request_source.class != Danger::RequestSources::BitbucketServer
+      super
+    end
+
+    # The instance name used in the Dangerfile
+    # @return [String]
+    #
+    def self.instance_name
+      "bitbucket_server"
+    end
+
+    def initialize(dangerfile)
+      super(dangerfile)
+      @bs = dangerfile.env.request_source
+    end
+
+    # @!group Bitbucket Server Misc
+    # The hash that represents the PR's JSON. For an example of what this looks like
+    # see the [Danger Fixture'd one](https://raw.githubusercontent.com/danger/danger/master/spec/fixtures/bitbucket_server_api/pr_response.json).
+    # @return [Hash]
+    def pr_json
+      @bs.pr_json
+    end
+
+    # @!group PR Metadata
+    # The title of the Pull Request.
+    # @return [String]
+    #
+    def pr_title
+      @bs.pr_json[:title].to_s
+    end
+
+    # @!group PR Metadata
+    # The body text of the Pull Request.
+    # @return [String]
+    #
+    def pr_description
+      @bs.pr_json[:description].to_s
+    end
+    alias pr_body pr_description
+
+    # @!group PR Metadata
+    # The username of the author of the Pull Request.
+    # @return [String]
+    #
+    def pr_author
+      @bs.pr_json[:author][:user][:slug].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The branch to which the PR is going to be merged into.
+    # @return [String]
+    #
+    def branch_for_base
+      @bs.pr_json[:toRef][:displayId].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # A href that represents the current PR
+    # @return [String]
+    #
+    def pr_link
+      @bs.pr_json[:links][:self].flat_map { |l| l[:href] }.first.to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The branch to which the PR is going to be merged from.
+    # @return [String]
+    #
+    def branch_for_head
+      @bs.pr_json[:fromRef][:displayId].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The base commit to which the PR is going to be merged as a parent.
+    # @return [String]
+    #
+    def base_commit
+      @bs.pr_json[:toRef][:latestCommit].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The head commit to which the PR is requesting to be merged from.
+    # @return [String]
+    #
+    def head_commit
+      @bs.pr_json[:fromRef][:latestCommit].to_s
+    end
+
+    # @!group Bitbucket Server Misc
+    # Returns a list of HTML anchors for a file, or files in the head repository.
+    # It returns a string of multiple anchors if passed an array.
+    # @param    [String or Array<String>] paths
+    #           A list of strings to convert to github anchors
+    # @param    [Bool] full_path
+    #           Shows the full path as the link's text, defaults to `true`.
+    #
+    # @return [String]
+    #
+    def html_link(paths, full_path: true)
```

Comment:
HTML links makes no sense here because bitbucket server supports markdown comments only.
We should rename this to `markdown_link` and generate markdown following this [guide](https://confluence.atlassian.com/bitbucketserver/markdown-syntax-guide-776639995.html#Markdownsyntaxguide-Linkstoexternalwebsites).